lib/galaxy/datatypes/binary.py

```diff
@@ -327,7 +257,11 @@ def to_archive(self, trans, dataset, name=""):
 
     def get_chunk(self, trans, dataset, offset=0, ck_size=None):
         index_file = dataset.metadata.bam_index
-        with pysam.AlignmentFile(dataset.file_name, "rb", index_filename=index_file.file_name) as bamfile:
+        if index_file:
```

Comment:
We don't actually need the index at all for `get_chunk`, so we should probably drop lines 259-263.